File: 参考/comp9041-Q/cs9041/lab07/log_probability.pl
#!/usr/bin/perl -w
$ori_word = shift @ARGV;
# print $ori_word;
$word = uc $ori_word;
# print $word,"\n";
foreach $file (glob "lyrics/*.txt") {
	$count_all = 0;
	$count = 0;
	$name = $file;
	$name =~ s/^[^\/]*\///;
	$name =~ s/\.txt$//;
	$name =~ tr/_/ /;
	# print $name,"\n";
	open F ,'<',$file or die;
	while ($line = <F>){
		@arr = $line =~ /[a-zA-Z]+/g;
		# $tmp = 0;
		foreach $x (@arr){
			if ($x){
				$count_all++;
			}

			chomp $x;
			$up = uc $x;
			# print $x,",",$up,"|";
			if ($up eq $word){
				$count++;
			}
		}
	}
	$new_count = $count+1;
	# $count = sprintf "%4d",$count;
	# $count_all = sprintf "%6d",$count_all;
	# $ans = sprintf "%.9f",$count/$count_all;
	# print "$count/$count_all = $ans $name","\n";
	printf "log((%d+1)/%6d) = %8.4f %s\n",$count,$count_all,log($new_count/$count_all),$name;
	close F;
}

File: 参考/comp9041-Q/cs9041/lab07/log_probability.py
#!/usr/bin/python3
import re,sys
from math import *
import glob
d={}
word = sys.argv[1]
for file in glob.glob("lyrics/*.txt"):
	name = re.sub('_',' ',file[7:-4])
	a = open(file).readlines()
	sum = 0
	word_sum = 0
	for line in a:
		l = re.split(r'[^a-zA-Z]',line)
		for x in l:
			if x:
				sum+=1
			if x.lower() == word.lower():
				word_sum += 1
	d[name] = (word_sum,sum)


for n in sorted(d.keys()):
	# print (n)
	print ("log(({}+1)/ {:6d}) = {:8.4f} {}".format(d[n][0],d[n][1],log((d[n][0]+1)/d[n][1]),n))

	    
    